src/SFML/Graphics/LineShape.cpp

```diff
@@ -0,0 +1,95 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/LineShape.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+LineShape::LineShape(const Vector2f& point1, const Vector2f& point2)
+    : m_direction(point2 - point1)
+    , m_thickness(2.f)
+{
+    assert(point1 != point2);
```

Comment:
I want to remove this assertion but it will require an additional condition in `getPoint` to ensure we're not calling `m_direction.normalize()` when `m_direction` has a length of zero which is an undefined operation. That may or may not have an acceptable performance penalty. I believe `getPoint` is called particularly often when drawing the shape so I wanted to keep it as efficient as possible.

Reply:
Is the call to normalise necessary? Can we not cache the points? Even with the assert the crash would still need to be prevented

Reply:
> Is the call to normalise necessary?

I believe so.

> Can we not cache the points? 

This should be possible.

> Even with the assert the crash would still need to be prevented

I added a condition that checks if the direction vector has zero length before trying to normalize it so that we can remove that assertion and allow a `LineShape` to have zero length.

Reply:
I honestly think the impact of assertions is overblown -- even more so because they can be disabled. But even if they couldn't, good diagnostics are often worth more than saving a tiny bit of performance in a (often non-critical) branch.

So in my opinion, please add assertions, if they help make code clearer and catch potential bugs :)